src/app/components/Ionic/IonicProvider.tsx

```diff
@@ -0,0 +1,38 @@
+import { createContext, FC, PropsWithChildren, useEffect } from 'react'
+import { useNavigate } from 'react-router-dom'
+import { Capacitor } from '@capacitor/core'
+import { App } from '@capacitor/app'
+
+const IonicContext = createContext<undefined>(undefined)
+
+const IonicContextProvider: FC<PropsWithChildren> = ({ children }) => {
+  const navigate = useNavigate()
+
+  useEffect(() => {
+    /**
+     * The back button refers to the physical back button on an Android device and should not be confused
+     * with either the browser back button or ion-back-button.
+     */
+    App.addListener('backButton', ({ canGoBack }) => {
+      if (!canGoBack) {
+        App.exitApp()
+      } else {
+        navigate(-1)
+      }
+    })
+
+    return () => {
+      App.removeAllListeners()
```

Comment:
App.addListener returns a way to remove that specific listener

Reply:
Nice, yes I was wondering how to do that. Will fix, so it only removes `backButton` listener.